src/lakefs_spec/spec.py

```diff
@@ -444,12 +449,68 @@ def _open(
             **kwargs,
         )
 
+    def put_file_to_blockstore(
+        self, lpath, repository, branch, resource, presign=True, storage_options=None
+    ):
+        blockstore_type = get_blockstore_type(self.client)
+        if blockstore_type not in ["s3", "gs", "azure"]:
+            raise ValueError(
+                f"Blockstore writes not implemented for blockstore type '{blockstore_type}'"
+            )
+
+        staging_location = self.client.staging_api.get_physical_address(
+            repository, branch, resource, presign=presign
+        )
+
+        if presign:
+            remote_url = staging_location.presigned_url
+            content_type, _ = mimetypes.guess_type(lpath)
+            if content_type is None:
+                content_type = "application/octet-stream"
+            with open(lpath, "rb") as f:
+                headers = {
+                    "Content-Type": content_type,
+                }
+                request = urllib.request.Request(
+                    url=remote_url, data=f, headers=headers, method="PUT"
+                )
+                try:
+                    if not remote_url.lower().startswith("http"):
+                        raise ValueError("Wrong protocol for remote connection")
+                    else:
+                        logger.info(f"Begin upload of {lpath}")
+                        with urllib.request.urlopen(
+                            request
+                        ):  # nosec [B310:blacklist] # We catch faulty protocols above.
+                            logger.info(f"Successfully uploaded {lpath}")
+                except urllib.error.HTTPError as e:
+                    urllib_http_error_as_lakefs_api_exception = ApiException(
+                        status=e.code, reason=e.reason
+                    )
+                    translate_lakefs_error(error=urllib_http_error_as_lakefs_api_exception)
+        else:
+            remote_url = staging_location.physical_address
+            remote = filesystem(blockstore_type, **(storage_options or {}))
```

Comment:
Follow-up note: The lakeFS blockstore name is `azure`, the fsspec registry entry is `az`; so we should add a check along the lines of

```python
if blockstore_type == "azure":
    blockstore_type = "az"
```